applications/vanilla/controllers/class.vanillasettingscontroller.php

```diff
@@ -83,15 +86,21 @@ public function posting() {
 
         // If seeing the form for the first time...
         if ($this->Form->authenticatedPostBack() === false) {
-
-
             // Apply the config settings to the form.
             $this->Form->setData($configurationModel->Data);
         } else {
             // This is a "reverse" field on the form. Disabling URL embeds is associated with a toggle that enables them.
             $disableUrlEmbeds = $this->Form->getFormValue('Garden.Format.DisableUrlEmbeds', true);
             $this->Form->setFormValue('Garden.Format.DisableUrlEmbeds', !$disableUrlEmbeds);
 
+            if ($this->Form->_FormValues['Vanilla.Comment.MaxLength'] > self::MAX_POST_LENGTH) {
+                $this->Form->addError(
+                    'Max comment length of' .
+                    $this->Form->_FormValues['Vanilla.Comment.MaxLength'] .
+                    ' value of is greater than the allowed size of' . self::MAX_POST_LENGTH . '.'
```

Comment:
"value of is greater than the allowed size of" isn't even grammatically correct, and it's generally ill-advisable to echo back user values to them (it's self-apparent when they typed in the form, it makes translations more difficult, and it can open a reflected XSS vector).

Reply:
"The highest allowed limit is X." would be a fine message.